pallets/orml-currencies-allowance-ext/src/lib.rs

```diff
@@ -0,0 +1,207 @@
+// #![deny(warnings)]
+#![cfg_attr(test, feature(proc_macro_hygiene))]
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(test)]
+extern crate mocktopus;
+
+use frame_support::{
+	dispatch::{DispatchResult},
+};
+#[cfg(test)]
+use mocktopus::macros::mockable;
+use orml_traits::MultiCurrency;
+use sp_runtime::{traits::*};
+use sp_std::{convert::TryInto, prelude::*, vec};
+
+pub use pallet::*;
+
+pub(crate) type BalanceOf<T> =
+	<<T as orml_currencies::Config>::MultiCurrency as orml_traits::MultiCurrency<
+		<T as frame_system::Config>::AccountId,
+	>>::Balance;
+
+pub(crate) type CurrencyOf<T> =
+	<<T as orml_currencies::Config>::MultiCurrency as orml_traits::MultiCurrency<
+		<T as frame_system::Config>::AccountId,
+	>>::CurrencyId;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use frame_support::pallet_prelude::*;
+
+	use super::*;
+
+	/// ## Configuration
+	/// The pallet's configuration trait.
+	#[pallet::config]
+	pub trait Config: frame_system::Config + orml_tokens::Config + orml_currencies::Config {
+		/// The overarching event type.
+		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+	}
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config> {
+		// RecoverFromErrors { new_status: StatusCode, cleared_errors: Vec<ErrorCode> },
+		UpdateActiveBlock {
+			block_number: T::BlockNumber,
+		},
+		/// (Additional) funds have been approved for transfer to a destination account.
+		ApprovedTransfer {
+			currency_id: CurrencyOf<T>,
+			source: T::AccountId,
+			delegate: T::AccountId,
+			amount: BalanceOf<T>,
+		},
+	}
+
+	#[pallet::error]
+	pub enum Error<T> {
+		/// Parachain is not running.
+		Unapproved,
+		ParachainNotRunning,
+		CurrencyNotLive,
+	}
+
+	#[pallet::storage]
+	/// Approved balance transfers. Balance is the amount approved for transfer.
+	/// First key is the asset ID, second key is the owner and third key is the delegate.
```

Comment:
Maybe move the comment above the macro statement. Not that important but makes it consistent with how we do it elsewhere.

Reply:
Agree. It is better to move comment up, and set above macro

Reply:
Done